gen.go

```diff
@@ -470,3 +478,19 @@ func applyFormats(schema *kin.Schema, obj formatable) {
 		prop.Value.WithFormat(fmt)
 	}
 }
+
+func applyEnums(schema *kin.Schema, obj enumerable) {
+	enums := obj.Enums()
+	for k, prop := range schema.Properties {
+		enum := enums[k]
+		if len(enum) == 0 {
+			continue
+		}
+
+		if prop.Value == nil {
+			continue
+		}
```

Comment:
Simpler and still clear IMO.
```suggestion
		if len(enum) == 0 || prop.Value == nil {
			continue
		}
```